### ISSUE TITLE
IPython.core.display:Pretty – wrong implementation

### ISSUE THREAD

**Issue by Carreau** [ipython/ipython#10595]
```
class Pretty(TextDisplayObject):

    def _repr_pretty_(self):
        return self.data
```

`_repr_pretty_`  should take 3 arguments: `_repr_pretty_(self, pp, cycle)`

**Comment by Carreau** [comment]
And should send the text to `pp.text(...)`

**Comment by kr1m** [comment]
I'll take this one

**Comment by Carreau** [comment]
Great ! Thanks !

**Comment by Carreau** [comment]
Closed by #10625.
